src/Compiler/CompilerUtils.cpp

```diff
@@ -152,90 +153,82 @@ static std::string getToolPath(std::string tool) {
     return std::string();
 }
 
-// Helper struct to make command construction and execution easy & readable.
-struct Command {
-  std::string _path;
-  std::vector<std::string> _args;
+// Append a single string argument.
+Command &Command::appendStr(const std::string &arg) {
+  if (arg.size() > 0)
+    _args.emplace_back(arg);
+  return *this;
+}
 
-  Command(std::string exePath)
-      : _path(std::move(exePath)),
-        _args({llvm::sys::path::filename(_path).str()}) {}
+// Append a single optional string argument.
+Command &Command::appendStrOpt(const llvm::Optional<std::string> &arg) {
+  if (arg.hasValue())
+    _args.emplace_back(arg.getValue());
```

Comment:
@Jiaqing-ASU can you please change `hasValue` and `getValue` to `has_value` and `value` instead? that will silence a deprecation compiler warning (which we've been working to weed out across the code base)

Reply:
see [this recent fix](https://github.com/onnx/onnx-mlir/pull/1622/files#diff-34127a063216e23b47bcb21b2b8dfe936b159241c0774480d4501593d5224798R173-R174)